### PR TITLE
Fixes: Build.psm1

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Fix typos
* Refactor code for readability and robustness
* Fix PSUseCmdletCorrectly rule violations

## PR Context

During #11563 issues were uncovered, hence this PR.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
- **User-facing changes**
    - [x] Not Applicable
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Comment by TravisEz13** [comment]
Was this tested on Windows PowerShell?

File: build.psm1 [hunk 1]
```diff
@@ -99,18 +99,16 @@ function Get-PSCommitId
 
 function Get-EnvironmentInformation
 {
-    $environment = @{}
+    $environment = @{'IsWindows' = [System.Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT}
```
Comment (TravisEz13) [inline build.psm1]: Has this been tested on Windows PowerShell?

File: build.psm1 [hunk 2]
```diff
@@ -99,18 +99,16 @@ function Get-PSCommitId
...
     # PowerShell will likely not be built on pre-1709 nanoserver
-    if ($PSVersionTable.ContainsKey("PSEdition") -and "Core" -eq $PSVersionTable.PSEdition) {
-        $environment += @{'IsCoreCLR' = $true}
-        $environment += @{'IsLinux' = $IsLinux}
-        $environment += @{'IsMacOS' = $IsMacOS}
-        $environment += @{'IsWindows' = $IsWindows}
+    if ('System.Management.Automation.Platform' -as [type]) {
```
Comment (TravisEz13) [inline build.psm1]: Has this been tested on Windows PowerShell?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: build.psm1 (hunk 2 above)
Comment (xtqqczze) [reply build.psm1]: I believe `PowerShell-CI-windows` build runs on Windows PowerShell
```
2020-01-13T20:20:50.2380725Z ========================== Starting Command Output ===========================
2020-01-13T20:20:50.2632322Z ##[command]"C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\a270093b-9d0f-4692-abef-11227601b6c1.ps1'"
```
https://dev.azure.com/powershell/2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_apis/build/builds/43352/logs/16

Comment (xtqqczze) [reply build.psm1]: And I also tested locally.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**